pandas/tests/test_expressions.py

```diff
@@ -380,3 +380,33 @@ def test_frame_series_axis(self, axis, arith):
 
         result = op_func(other, axis=axis)
         tm.assert_frame_equal(expected, result)
+
+    @pytest.mark.parametrize(
+        "op",
+        [
+            "__mod__",
+            pytest.param("__rmod__", marks=pytest.mark.xfail(reason="GH-36552")),
+            "__floordiv__",
+            "__rfloordiv__",
+        ],
+    )
+    @pytest.mark.parametrize(
+        "box, tester",
+        [
+            (DataFrame, tm.assert_frame_equal),
+            (Series, tm.assert_series_equal),
+            (Index, tm.assert_index_equal),
+        ],
+    )
+    @pytest.mark.parametrize("scalar", [-5, 5])
+    def test_python_semantics_with_numexpr_installed(self, op, box, tester, scalar):
+        # https://github.com/pandas-dev/pandas/issues/36047
+        expr._MIN_ELEMENTS = 0
+        data = np.arange(-50, 50)
+        obj = box(data)
+        method = getattr(obj, op)
+        result = method(scalar)
+        expr.set_use_numexpr(False)
+        expected = method(scalar)
+        expr.set_use_numexpr(True)
+        tester(result, expected)
```

Comment:
Can you also test against a manually constructed expected result? 
(or do you know if we have elsewhere a test that covers this behaviour with negative numbers for modulo ?)

Reply:
i'll look though the tests again to see what we already have.

the method i used here is inspired by `run_arithmetic`, `run_binary` and also `run_frame`

I kept the object sizes small here so could do elementwise comparion similar to code in https://github.com/pandas-dev/pandas/pull/36552#discussion_r493281848 wdyt?

Reply:
It's certainly good to test the equivalency of numpy vs numexpr behaviour. But here we discovered a specific case where they are not equal, and I think it is useful to also (in addition, can also be a separate test) test this corner case of negative values with modulo. 
We might change the implementation in the future (eg assume we would take on numexpr as required dependency and always use it), and then we wouldn't necessarily catch this case.

Reply:
> The object size shouldn't matter

I meant iterating through in Python was not a resource issue.

I have added an element-wise comparison but tests are failing at the moment

```
>>> pd.__version__
'1.2.0.dev0+499.g3b12293cea'
>>>
>>> 5 // 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ZeroDivisionError: integer division or modulo by zero
>>>
>>> 5 // np.array([0, 0])[0]
<stdin>:1: RuntimeWarning: divide by zero encountered in long_scalars
0
>>>
>>> 5 // np.array(0)
<stdin>:1: RuntimeWarning: divide by zero encountered in floor_divide
0
>>>
>>> 5 // np.array([-1, 0, 1])
array([-5,  0,  5], dtype=int32)
>>>
>>> 5 // pd.Series([-1, 0, 1])
0   -5.0
1    inf
2    5.0
dtype: float64
>>>
>>> -5 // np.array([-1, 0, 1])
array([ 5,  0, -5], dtype=int32)
>>>
>>> -5 // pd.Series([-1, 0, 1])
0    5.0
1   -inf
2   -5.0
dtype: float64
>>>
```

Reply:
fixed test. this discrepancy that has surfaced is not related to the regression issue, will investigate further.

Reply:
I think pandas is doing the right thing here? other than returning a non-integer result for floordiv. maybe should return IntegerArray in future

```
>>> 5 / np.array([-1, 0, 1])
<stdin>:1: RuntimeWarning: divide by zero encountered in true_divide
array([-5., inf,  5.])
>>>
>>> 5 / pd.Series([-1, 0, 1])
0   -5.0
1    inf
2    5.0
dtype: float64
>>>
```


> Can you also test against a manually constructed expected result?

still outstanding